src/coreclr/gc/gc.cpp

```diff
@@ -21174,10 +21174,9 @@ uint8_t* gc_heap::find_object (uint8_t* interior)
         heap_segment* seg = find_segment (interior, FALSE);
         if (seg)
         {
-#ifdef FEATURE_CONSERVATIVE_GC
             if (interior >= heap_segment_allocated(seg))
                 return 0;
-#endif
```

Comment:
functionally I don't see a problem. I'm curious - could interior be actually >= heap_segment_allocated if GCConfig::GetConservativeGC() is false? this would mean it's reporting an interior pointer that's indeed on a segment but not pointing at a valid object on that seg.